tests/test_registration.py

```diff
@@ -160,26 +162,157 @@ def test_mse_decomposition(self):
+        deltas = reg.deltas_.round(3)
+        np.testing.assert_array_almost_equal(deltas, [-0.022, 0.03])
 
-    def test_shift_registration(self):
+        # Test with Basis
+        fd = self.fd.to_basis(Fourier())
+        reg.fit_transform(fd)
+        deltas = reg.deltas_.round(3)
+        np.testing.assert_array_almost_equal(deltas, [-0.022, 0.03])
+
+    def test_fit_and_transform(self):
         """Test wrapper of shift_registration_deltas"""
 
-        fd = make_sinusoidal_process(n_samples=2, error_std=0, random_state=1)
+        fd = make_sinusoidal_process(n_samples=2, error_std=0, random_state=10)
+
+        reg = ShiftRegistration()
+        response = reg.fit(self.fd)
+
+        # Check attributes and returned value
+        self.assertTrue(hasattr(reg, 'template_'))
+        self.assertTrue(response is reg)
+
+        fd_registered = reg.transform(fd)
+        deltas = reg.deltas_.round(3)
+        np.testing.assert_array_almost_equal(deltas, [0.071, -0.071])
+
+    def test_inverse_transform(self):
+
+        reg = ShiftRegistration()
+        fd = reg.fit_transform(self.fd)
+        fd = reg.inverse_transform(fd)
+
+        np.testing.assert_array_almost_equal(fd.data_matrix,
+                                             self.fd.data_matrix, decimal=3)
+
+    def test_raises(self):
+
+        reg = ShiftRegistration()
+
+        # Test not fitted
+        with np.testing.assert_raises(NotFittedError):
+            reg.transform(self.fd)
+
+        reg.fit(self.fd)
+        reg.set_params(restrict_domain=True)
+
+        # Test use fit or transform with restrict_domain=True
+        with np.testing.assert_raises(AttributeError):
+            reg.transform(self.fd)
+
+        with np.testing.assert_raises(AttributeError):
+            reg.fit(self.fd)
+
+        # Test inverse_transform without previous transformation
+        with np.testing.assert_raises(AttributeError):
+            reg.inverse_transform(self.fd)
+
+        reg.fit_transform(self.fd)
+
+        # Test inverse transform with different number of sample
+        with np.testing.assert_raises(ValueError):
+            reg.inverse_transform(self.fd[:1])
+
+        fd = make_multimodal_samples(dim_domain=2, random_state=0)
+
+        with np.testing.assert_raises(NotImplementedError):
+            reg.fit_transform(fd)
+
+        reg.set_params(initial=[0.])
+
+        # Wrong initial estimation
+        with np.testing.assert_raises(ValueError):
+            reg.fit_transform(self.fd)
+
+    def test_template(self):
+
+        reg = ShiftRegistration()
+        fd_registered_1 = reg.fit_transform(self.fd)
+
+        reg_2 = ShiftRegistration(template=reg.template_)
+        fd_registered_2 = reg_2.fit_transform(self.fd)
+
+        reg_3 = ShiftRegistration(template=mean)
+        fd_registered_3 = reg_3.fit_transform(self.fd)
+
+        reg_4 = ShiftRegistration(template=reg.template_)
+        fd_registered_4 = reg_4.fit(self.fd).transform(self.fd)
```

Comment:
Use `fit_transform`.

Reply:
The fit_transform method is tested six lines above. This test case is made to test the usage of fit with a templated fixed and the following transformation.

Reply:
Ok.